charms/reactive/relations.py

```diff
@@ -267,6 +272,16 @@ def changed(self):
     recommended that this be used only with :attr:`scopes.GLOBAL` scope.
     """
 
+    @classmethod
+    def _startup(cls):
+        # update data to be backwards compatible after fix for issue 28
+        _migrate_conversations()
```

Comment:
Similarly here, if _migrate_conversations isn't usually a noop we shouldn't call it in restricted_mode.